### DISCUSSION TITLE
[FEATURE REQUEST]: Add AwaitTerminationAsync for StreamingQuery

### DISCUSSION THREAD

**Discussion by SWATOPLUS** [dotnet/spark#775] [Q&A]
I am wanna to mange with many StreamingQuery in single spark job. So adding a AwaitTerminationAsync method should resolve this problem. 

The example solution

```C#
        private static async Task AwaitTerminationAsync(StreamingQuery query, CancellationToken token)
        {
            while (query.IsActive())
            {
                await Task.Delay(100, token);

                if (token.IsCancellationRequested)
                {
                    query.Stop();
                }
            }

            var ex = query.Exception();

            if (ex != null)
            {
                throw ex;
            }
        }
```

**Comment by imback82** [comment]
@SWATOPLUS we only expose APIs that are already available on the Java/Scala side. This can be just added in your application as needed?

**Comment by SWATOPLUS** [comment]
I think it will be good if you found some other way to complete the task, instead of Delay. Maybe java api has some callback registration. and so on. Yeah this is not hottest feature, but it need to be implemented in dotnet way.
